port_ocean/core/integrations/mixins/sync_raw.py

```diff
@@ -233,13 +250,43 @@ async def register_raw(
             resource for resource in config.resources if resource.kind == kind
         ]
 
-        return await asyncio.gather(
+        diffs: list[EntitySelectorDiff] = await asyncio.gather(
             *(
                 self._register_resource_raw(resource, results, user_agent_type, True)
                 for resource in resource_mappings
             )
         )
 
+        registered_entities: list[Entity] = [
+            item for d in diffs for item in d["passed"]
+        ]
+        passed_identifiers_blueprints = set(
+            (item.identifier, item.blueprint) for d in diffs for item in d["passed"]
+        )
+
+        entities_to_delete = [
+            item
+            for d in diffs
+            for item in d["failed"]
+            if (item.identifier, item.blueprint) not in passed_identifiers_blueprints
+        ]
+
+        filtered_entities_to_delete: list[Entity] = await self._search_deleted_entities(
+            user_agent_type, entities_to_delete
+        )
+
+        if len(entities_to_delete) > len(filtered_entities_to_delete):
+            logger.info(
+                f"Skipping deletion for {len(entities_to_delete) - len(filtered_entities_to_delete)} entities "
+                f"as we couldn't find matching entities that's related to the current integration."
+            )
```

Comment:
could be very spamming message, as it will happen most of the time probably. maybe just add a log before the deletion, that you are going to perform a delete